gamemode/modules/jobs/sv_jobs.lua

```diff
@@ -137,26 +137,41 @@ function meta:changeTeam(t, force, suppressNotification, ignoreMaxMembers)
         DarkRP.resetLaws()
     end
 
+    local DoEffect = false
+
     self:SetTeam(t)
     hook.Call("OnPlayerChangedTeam", GAMEMODE, self, prevTeam, t)
     DarkRP.log(self:Nick() .. " (" .. self:SteamID() .. ") changed to " .. team.GetName(t), nil, Color(100, 0, 255))
     if self:InVehicle() then self:ExitVehicle() end
     if GAMEMODE.Config.norespawn and self:Alive() then
         self:StripWeapons()
         self:RemoveAllAmmo()
+
+        DoEffect = true
+        player_manager.SetPlayerClass(self, TEAM.playerClass or "player_darkrp")
+        self:applyPlayerClassVars(false)
+        gamemode.Call("PlayerSetModel", self)
+        gamemode.Call("PlayerLoadout", self)
+    else
+        if GAMEMODE.Config.instantjob then
+            DoEffect = true
+
+            self:StripWeapons()
+            self:RemoveAllAmmo()
+            self:Spawn()
+        else
+            self:KillSilent()
+        end
+    end
+
+    if DoEffect then
```

Comment:
Wouldn't this effect be run _after_ the player has spawned on line 161? The effect would appear at the player's spawn position, rather than where they were.

Reply:
I made it like that on purpose as I feel it looks better spawning in with the effect, I can change it if you like.